File: SingleNeuronClassification/src/main/java/Network.java
import java.util.List;

public class Network {
    private Neuron neuron;
    private int inputCount;
    private double learningRate;

    public Network(int inputCount) {
        neuron = new Neuron(inputCount);
        this.inputCount = inputCount;
        learningRate = 0.01;
    }

    public double work(List<Double> inputs) {
        double result = 0;
        for (int i = 0; i < inputCount; i++) {
            result += neuron.getWeigths().get(i) * inputs.get(i);
        }
        return result;
    }

    public void adapt(List<Double> inputs, double desiredResult) {
        double result = work(inputs);
        double resultOffset = desiredResult - result;

        for (int i = 0; i < inputCount; i++) {
            double weightChange = resultOffset * -inputs.get(i);
            neuron.getWeigths().set(i, neuron.getWeigths().get(i) - learningRate * weightChange);
        }
    }
}

File: SingleNeuronClassificationPY/NetworkComponents/Network.py
from NetworkComponents import Neuron


class Network:
    def __init__(self, input_count):
        self.neuron = Neuron.Neuron(input_count)
        self.input_count = input_count
        self.learning_rate = 0.01

    def work(self, inputs):
        result = 0
        for i in range(len(inputs)):
            result += self.neuron.get_weights()[i] * inputs[i]
        if result > 0.5:
            return 1
        else:
            return 0

    def adapt(self, example):
        result = self.work(example.getInputs())
        result_offset = example.desired_result - result

        for i in range(self.input_count):
            weight_change = result_offset * (- example.getInputs()[i])
            self.neuron.get_weights()[i] -= self.learning_rate * weight_change
